espnet2/fst/lm_rescore.py

```diff
@@ -0,0 +1,569 @@
+from typing import Dict
+from typing import List
+
+import k2
+import torch
+
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+
+def remove_repeated_and_leq(tokens: List[int], blank_id: int = 0):
+    """
+    Genrate valid token sequence.
+    Result may be used as input of transformer decoder and neural language model.
+    Fristly, remove repeated token from a "token alignment" seqs;
+    Then remove blank symbols.
+    This fuction may be replaced by tokenizing word_seqs with tokenizer
+    or composeing word_seqs_fsas with L_inv.fst
+    or composing token_seqs with ctc_topo.
+    Current method is chosed other than previous three methods because it won't need an extra object, i.e. tokenizer, L.fst or ctc_topo.
+    """
+    new_tokens = []
+    previous = None
+    for token in tokens:
+        if token != previous:
+            new_tokens.append(token)
+            previous = token
+    new_tokens = [token for token in new_tokens if token > blank_id]
+    return new_tokens
+
+
+def compute_am_scores_and_fm_scores(
+    lats: k2.Fsa, word_fsas_with_epsilon_loops: k2.Fsa, path_to_seq_map: torch.Tensor
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    """Compute AM and LM scores of n-best lists (represented as word_fsas).
```

Comment:
I think the AM score is not correct terminology.
AM should be replaced with e2eASR or something like that?

Reply:
If it requires too many changes or not aligned with k2’s terminology or we only limit this use for CTC, of course you can keep “am”. 
But I think it’s better to add a note about this terminology.